File: SpearoGo/Views/FishActivityPage.swift
import SwiftUI

struct FishActivityPage: View {
    @Environment(AppState.self) private var appState
    @State private var crownOffset: Double = 0

    var body: some View {
        ScrollView {
            VStack(spacing: Brand.Spacing.section) {
                Text("Fish Activity")
                    .brandSectionHeader()

                if let sol = appState.solunarData {
                    let moon = moonInfo(sol.moonPhase)
                    Grid(alignment: .center, horizontalSpacing: 20, verticalSpacing: Brand.Spacing.section) {
                        GridRow {
                            ConditionItem(icon: moon.icon,
                                          label: "Moon",
                                          value: String(format: "%.0f%%", sol.moonIllumination * 100),
                                          unit: "")
                            ConditionItem(icon: "fish.fill",
                                          label: "Rating",
                                          value: sol.activityRating,
                                          unit: "")
                        }
                    }
                    .accessibilityElement(children: .combine)
                    .accessibilityLabel(moonAccessibilityLabel(sol))

                    VStack(spacing: Brand.Spacing.micro) {
                        if let major = sol.nextMajorPeriod {
                            SolunarPeriodRow(label: "Major", time: major, color: Brand.Colors.go)
                        }
                        if let minor = sol.nextMinorPeriod {
                            SolunarPeriodRow(label: "Minor", time: minor, color: Brand.Colors.maybe)
                        }
                    }
                } else {
                    // Shimmer skeleton
                    Grid(alignment: .center, horizontalSpacing: 20, verticalSpacing: Brand.Spacing.section) {
                        GridRow {
                            ConditionItemSkeleton()
                            ConditionItemSkeleton()
                        }
                    }
                    VStack(spacing: Brand.Spacing.micro) {
                        SkeletonBlock(width: 140, height: 26)
                        SkeletonBlock(width: 140, height: 26)
                    }
                    .accessibilityLabel("Loading fish activity data")
                }
            }
            .padding(Brand.Spacing.page)
        }
        .focusable()
        .digitalCrownRotation($crownOffset)
        .brandPage()
    }

    private func moonInfo(_ phase: Double) -> (icon: String, name: String) {
        switch phase {
        case 0..<0.1, 0.9...: return ("moonphase.new.moon",          "New Moon")
        case 0.1..<0.25:      return ("moonphase.waxing.crescent",   "Waxing Crescent")
        case 0.25..<0.35:     return ("moonphase.first.quarter",     "First Quarter")
        case 0.35..<0.5:      return ("moonphase.waxing.gibbous",    "Waxing Gibbous")
        case 0.5..<0.6:       return ("moonphase.full.moon",         "Full Moon")
        case 0.6..<0.75:      return ("moonphase.waning.gibbous",    "Waning Gibbous")
        case 0.75..<0.9:      return ("moonphase.last.quarter",      "Last Quarter")
        default:              return ("moonphase.waning.crescent",   "Waning Crescent")
        }
    }

    private func moonAccessibilityLabel(_ sol: SolunarData) -> String {
        let phase = moonInfo(sol.moonPhase).name
        let illum = String(format: "%.0f percent illumination", sol.moonIllumination * 100)
        return "\(phase), \(illum). Activity rating: \(sol.activityRating)"
    }
}

struct SolunarPeriodRow: View {
    let label: String
    let time: Date
    let color: Color

    var body: some View {
        HStack {
            Text(label)
                .font(Brand.Typography.itemLabel)
                .kerning(Brand.Kerning.itemLabel)
                .foregroundStyle(color)
                .frame(width: 40, alignment: .leading)

            Text(time, style: .time)
                .font(Brand.Typography.periodTime)
                .foregroundStyle(Brand.Colors.textPrimary)

            Spacer()

            Circle()
                .fill(color)
                .frame(width: 5, height: 5)
        }
        .padding(.horizontal, Brand.Spacing.page)
        .padding(.vertical, 4)
        .background(Brand.Colors.textPrimary.opacity(Brand.Opacity.cardFill))
        .clipShape(RoundedRectangle(cornerRadius: Brand.Radius.chip))
        .accessibilityElement(children: .combine)
        .accessibilityLabel("\(label) period")
    }
}

// MARK: - Previews

#Preview {
    FishActivityPage()
        .previewAsWatch()
        .environment(AppState.preview())
}

#Preview("Loading") {
    FishActivityPage()
        .previewAsWatch()
        .environment(AppState.previewLoading())
}

File: wear/app/src/main/java/com/spearotracker/spearogo/ui/pages/FishActivityPage.kt
package com.spearotracker.spearogo.ui.pages

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.wear.compose.material3.Text
import com.spearotracker.spearogo.ui.AppUiState
import com.spearotracker.spearogo.ui.components.ConditionItem
import com.spearotracker.spearogo.ui.components.ConditionItemSkeleton
import com.spearotracker.spearogo.ui.theme.Brand
import java.text.SimpleDateFormat
import java.util.*

@Composable
fun FishActivityPage(uiState: AppUiState) {
    val scrollState = rememberScrollState()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(scrollState)
            .padding(Brand.Spacing.page),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Text(
            text = "FISH ACTIVITY",
            style = Brand.Typography.sectionHeader,
            color = Brand.Colors.textSecondary,
            modifier = Modifier.padding(bottom = Brand.Spacing.item)
        )

        val sol = uiState.solunarData

        if (sol != null) {
            val moonIcon = moonPhaseName(sol.moonPhase)

            Row(
                horizontalArrangement = Arrangement.spacedBy(20.dp),
                modifier = Modifier.padding(bottom = Brand.Spacing.section)
            ) {
                ConditionItem(icon = "moon", label = "Moon", value = "%.0f%%".format(sol.moonIllumination * 100), unit = "")
                ConditionItem(icon = "fish", label = "Rating", value = sol.activityRating, unit = "")
            }

            val timeFormat = SimpleDateFormat("HH:mm", Locale.getDefault())

            // Major period
            sol.nextMajorPeriod?.let { major ->
                SolunarPeriodRow(label = "Major", time = timeFormat.format(Date(major)), color = Brand.Colors.go)
            }
            // Minor period
            sol.nextMinorPeriod?.let { minor ->
                SolunarPeriodRow(label = "Minor", time = timeFormat.format(Date(minor)), color = Brand.Colors.maybe)
            }
        } else {
            Row(horizontalArrangement = Arrangement.spacedBy(20.dp), modifier = Modifier.padding(bottom = Brand.Spacing.section)) {
                ConditionItemSkeleton()
                ConditionItemSkeleton()
            }
        }
    }
}

@Composable
private fun SolunarPeriodRow(label: String, time: String, color: Color) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = Brand.Spacing.page, vertical = 4.dp)
            .background(
                Brand.Colors.textPrimary.copy(alpha = Brand.Opacity.cardFill),
                RoundedCornerShape(Brand.Radius.chip)
            )
            .padding(horizontal = Brand.Spacing.page, vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(
            text = label.uppercase(),
            style = Brand.Typography.itemLabel,
            color = color,
            modifier = Modifier.width(40.dp)
        )
        Text(
            text = time,
            style = Brand.Typography.periodTime,
            color = Brand.Colors.textPrimary
        )
        Spacer(modifier = Modifier.weight(1f))
        Box(
            modifier = Modifier
                .size(5.dp)
                .clip(CircleShape)
                .background(color)
        )
    }
}

private fun moonPhaseName(phase: Double): String = when {
    phase < 0.1 || phase >= 0.9 -> "New Moon"
    phase < 0.25 -> "Waxing Crescent"
    phase < 0.35 -> "First Quarter"
    phase < 0.5 -> "Waxing Gibbous"
    phase < 0.6 -> "Full Moon"
    phase < 0.75 -> "Waning Gibbous"
    phase < 0.9 -> "Last Quarter"
    else -> "Waning Crescent"
}
